lib/python/qmk/cli/compile.py

```diff
@@ -28,6 +29,25 @@ def compile(cli):
     If --keyboard and --keymap are provided this command will build a firmware based on that.
 
     """
+    # Set CWD as directory command was issued from
+    cwd = os.path.normpath(os.path.join(os.environ['ORIG_CWD']))
+
+    # Initialize boolean to check for being in a keyboard directory and initialize keyboard string
+    in_keyboard = False
+    keyboard = ""
+
+    # Loop through three levels up to check for keyboards root directory
+    for index, directory in enumerate(['..', '../..', '../../..']):
+        above_basename = os.path.basename(os.path.normpath(os.path.join(cwd, directory)))
+
+        # If the folder above is "/keyboards/"
+        if above_basename == "keyboards":
+            # Set the full path name and get back the full keyboard name
+            path = Path(os.path.normpath(os.path.join(cwd, directory[index-1])))
+            keyboard += str(path.relative_to(os.getcwd() + "/keyboards"))
```

Comment:
Maybe you should also put the QMK directory into an environment variable (in `bin/qmk`), implicitly using the current working directory seems error-prone to me..

Reply:
I agree, that was one part I thought may be wrong as well. I am happy to try that out if that's a better approach

Reply:
The `qmk` script always changes current directory to the top level of `qmk_firmware`. You can find the directory the user ran the script from in `os.environ['ORIG_CWD']`. When looking for paths within qmk_firmware it's cleanest to use relative paths.

Reply:
True, but imo that isn't very clean/obvious when looking at the code. Maybe a `QMK_DIR` environment variable being set in the `qmk` script would still be a good idea? This might be another PR though.

Reply:
I'm open to discussing that further, but at this point we're out of scope for this PR. Can we move this to a discord discussion or a new issue?